### PR TITLE
Refactored _rclpy.rclpy_get_rmw_qos_profile to return dictionary instead of QoSProfile

### PR DESCRIPTION
**Pull request by skucheria**
Fixes https://github.com/ros2/rclpy/issues/358

### CODE REVIEW

File: rclpy/src/rclpy_common/src/common.c [hunk 1]
```diff
@@ -266,6 +359,8 @@ rclpy_common_convert_to_py_qos_policy(const rmw_qos_profile_t * qos_profile)
     Py_DECREF(pyqos_profile_class);
     return NULL;
   }
+
```
Comment (jacobperron) [inline rclpy/src/rclpy_common/src/common.c]: nit: This change should be reverted.

File: rclpy/src/rclpy/_rclpy.c [hunk 2]
```diff
@@ -3557,19 +3557,19 @@ rclpy_get_rmw_qos_profile(PyObject * Py_UNUSED(self), PyObject * args)
   }
   PyObject * pyqos_profile = NULL;
   if (0 == strcmp(pyrmw_profile, "qos_profile_sensor_data")) {
-    pyqos_profile = rclpy_common_convert_to_py_qos_policy(&rmw_qos_profile_sensor_data);
```
Comment (jacobperron) [inline rclpy/src/rclpy/_rclpy.c]: If `rclpy_common_convert_to_py_policy` is not used anywhere else, I think we can remove it's definition in favor of `rclpy_common_convert_to_qos_dict`.

**Comment by skucheria** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7280/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3456/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5956/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7092/)

**Comment by jacobperron** [comment]
It looks like those test failures appear in the nightlies as well. You can squash and merge this change :+1:

**Pull request merged**